Fix bug in code example combobox.md

devbyray opened the pull request:
The default value of `const value = ref({})` should contain a falsy value in order to show the "Select framework...".

![image](https://github.com/radix-vue/shadcn-vue/assets/6241164/04496c48-bfb3-47a7-a48b-3dce5b1d4c22)

![image](https://github.com/radix-vue/shadcn-vue/assets/6241164/71df90ec-e0e2-4e0e-9a14-2587a1312e34)

When giving it an `{}` empty object will result in a truthy value which is not what we want. When given an a falsy value like `null` it will display the correct text in the button.

devbyray committed:
Update combobox.md

The default value of `const value = ref({})` should contain a falsy value in order to show the "Select framework...".

devbyray committed:
Fix is with showing the selected value

sadeghbarati commented on apps/www/src/content/docs/components/combobox.md:
```diff
@@ -50,7 +50,7 @@ const frameworks = [
 
 const open = ref(false)
-const value = ref({})
+const value = ref(null)
```
I think an empty string will do the trick

```suggestion
const value = ref('')
```

sadeghbarati commented on apps/www/src/content/docs/components/combobox.md:
```diff
@@ -62,7 +62,7 @@ const value = ref({})
         class="w-[200px] justify-between"
       >
-        {{ value ? frameworks.find((framework) => framework.value === value)?.label : 'Select framework...' }}
+        {{ value ? frameworks.find((framework) => framework.value === value?.value)?.label : 'Select framework...' }}
```
based on [`ComboboxDemo.vue`](https://github.com/radix-vue/shadcn-vue/blob/dev/apps/www/src/lib/registry/default/example/ComboboxDemo.vue)

There is no need to use `.value` in `<template>`

```suggestion
        {{ value ? frameworks.find((framework) => framework.value === value)?.label : 'Select framework...' }}
```

devbyray committed:
Set empty string to value instead of null

Co-authored-by: Sadegh Barati <sadeghbaratiwork@gmail.com>

devbyray committed:
Update apps/www/src/content/docs/components/combobox.md

Co-authored-by: Sadegh Barati <sadeghbaratiwork@gmail.com>

devbyray committed:
Update combobox.md

Since the selected `value` is an Object the check is not visible when comparing it with `value`, it should be `value?.value`.

devbyray commented:
@sadeghbarati thanks for your suggestions. It appears that the Codesandbox example is different then the one in the Markdown file: https://codesandbox.io/p/devbox/shadcn-vue-comboboxdemo-368j7c?file=%2Fsrc%2FApp.vue

The `:value="framework"` in the Markdown example uses the whole object as value. In the Codesandbox it is `:value="framework.value"`. 

![image](https://github.com/radix-vue/shadcn-vue/assets/6241164/67ddc097-46ab-4a8b-bc10-1fb33c71fde9)

sadeghbarati commented on apps/www/src/content/docs/components/combobox.md:
```diff
@@ -82,7 +82,7 @@ const value = ref({})
                 :class="cn(
                   'mr-2 h-4 w-4',
-                  value === framework.value ? 'opacity-100' : 'opacity-0',
+                  value?.value === framework.value ? 'opacity-100' : 'opacity-0',
```
```suggestion
                  value === framework.value ? 'opacity-100' : 'opacity-0',
```

sadeghbarati committed:
Update apps/www/src/content/docs/components/combobox.md

sadeghbarati committed:
chore: update

devbyray commented:
@sadeghbarati I'm to see the fix being merged 💪
